### PR TITLE
Use new cheeseshop alongside the old (for now)

### LINKED ISSUE #267: Problem bootstrapping Pants on Mac M1
A user reported this error:

```
Bootstrapping Pants 2.16.0 using cpython 3.9.16
Installing pantsbuild.pants==2.16.0 into a virtual environment at /Users/jan/Library/Caches/nce/aeded33e4c...e/bindings/venvs/2.16.0
  Preparing metadata (setup.py) ... done
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> See above for output.

  note: This error originates from a subprocess, and is likely not a problem with pip.
  Building wheel for psutil (setup.py) ... error
  ERROR: Failed building wheel for psutil
  error: subprocess-exited-with-error

  × Running setup.py install for psutil did not run successfully.
  │ exit code: 1
  ╰─> See above for output.

  note: This error originates from a subprocess, and is likely not a problem with pip.
  Running setup.py install for psutil ... error
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> psutil

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
Install failed: Command '['/Users/jan/Library/Caches/nce/aede...e/bindings/venvs/2.16.0/bin/python', '-sE', '-m', 'pip', '--disable-pip-version-check', '--no-python-version-warning', '--log', '/Users/jan/Library/Caches/nce/aede...bf2e/bindings/venvs/2.16.0/pants-install.log', 'install', '--quiet', '--find-links', 'https://wheels.pantsbuild.org/simple/', '--progress-bar', 'off', 'pantsbuild.pants==2.16.0']' returned non-zero exit status 1.
More information can be found in the log at: /Users/jan/Library/Caches/nce/aed...2e/bindings/logs/install.log

Error: Isolates your Pants from the elements.

Please select from the following boot commands:

scie-pants
bootstrap-tools
pants
pants-debug
update

You can select a boot command by passing it as the 1st argument or else by setting the SCIE_BOOT environment variable.

ERROR: Failed to establish atomic directory /Users/jan/Library/Caches/nce/aeded33e4c5a...e. Population of work directory failed: Boot binding command failed: exit status: 1
```

Context: 

https://pantsbuild.slack.com/archives/C046T6T9U/p1693997543934359

### PR DESCRIPTION
**Pull request by thejcannon**
Fixes #267 by reverting https://github.com/pantsbuild/scie-pants/pull/235, and then pivoting so that the new cheesehsop is added alongside the other links that were already being provided.

This makes it so `scie-pants` can still work on much older PyPI-yanked versions, and also work on Macs.

I consider this a stopgap until we backport the per-platform PEXs, in which case, all of v2 would just use the PEX.

### CODE REVIEW

File: tools/src/scie_pants/pants_release_tags.json [hunk 1]
```diff
@@ -366,5 +366,286 @@
...
+  "release_2.15.0.dev3": "a13175ea1d729e0b187fb158e584fc7764533584",
+  "release_2.15.0.dev4": "458c7f04e32cdfba61b2d803d65a077c7d0165fe",
+  "release_2.15.0.dev5": "a21ac88803801cd729e3fb5ff2c187831d708d73",
+  "release_2.15.0a0": "521ee5fe06556233dcaaf1eea4cfe1d564f63642",
+  "release_2.15.0a1": "a7d97014a9ef5626dfa92ab45814ca173f9b60da",
+  "release_2.15.0rc0": "54712e15fdee9ea6cb38e134761222eb3284a3a0",
+  "release_2.16.0.dev0": "c7395e4504738da024dc2514ef4bb9316359ab07",
+  "release_2.16.0.dev1": "57fecb6ce9eee75be20f28fc6c19feb572fa841d",
+  "release_2.16.0.dev2": "c43206feac8cf420c3f3c65fbb9929d75a46f4da",
+  "release_2.2.0": "81fd0314465d9d1607bd3bc9b36d608434236e75",
```
Comment (huonw) [inline tools/src/scie_pants/pants_release_tags.json:505]: Could you remind me if more releases of 2.16 and 2.17 (things before `2.18.0.dev5`) need to appear here?

**Review by huonw — APPROVED** [review]
Looks reasonable, although I don't fully understand the original problem (e.g. I've been using scie-pants on M1 Mac with 2.16 for quite a while...) nor understand all of the things that feed into this. 🤔

File: tools/src/scie_pants/pants_release_tags.json (hunk 1 above)
Comment (thejcannon) [reply tools/src/scie_pants/pants_release_tags.json:505]: This file is just a cache (or whatever the baked-in, static version of a cache is) which is checked first before making an unauthenticated GitHub request.

**Comment by thejcannon** [comment]
So the issue is wheels didnt exist for some of Pants dependencies for, I assume, a long while for Mac.

Pants solution to this was that during a release, on the build agent, compile the transitive set of dependencies into wheels, and store those off in S3 under a special path (using the sha of the tag).

Then, I assume, `./pants`, and therefore also `scie-pants` had the behavior of setup up the Pants venv by pointing to this list of wheels.

You may not see it if you're using a cached venv, or if you're able to build the wheels on your system.

File: tools/src/scie_pants/pants_release_tags.json (hunk 1 above)
Comment (thejcannon) [reply tools/src/scie_pants/pants_release_tags.json:505]: Honestly, I feel like if the rate limit errors were material we'd see them by now. I'm much more worried about getting the SHA wrong. 

If we really wanted to, is suggest we did that in a dedicated PR

**Review by benjyw — APPROVED** [review]

**Pull request merged**